### DISCUSSION TITLE
Ledvance CCT Downlight limit Color Temperature

### DISCUSSION THREAD

**Discussion by mirisbowring** [make-all/tuya-local#1259] [Q&A] [answered]
Hi,
previously I've used the "local tuya" integration but it was very unreliable and after some testing, this one seems to be much more reliable.
Today, i discovered a little Issue with my Downlights:
```
color_temp (6535) must be between 0 and 1000
```
I have Ledvance Sun@Home Downlights with a color temp from 2200K to 5000K.
When making a light group in HA and adding multiple of those lights from this integration, i get an error when sliding the "coldest" color. Seems like this integration thinks in 2700-6500K.

How can i change the color temparature space?

**Comment by make-all** [comment, marked as answer]
Probably the light group is defaulting to HA's default color temperature range, which is a bit wider than the 2700-6500 set for the generic cct_lightbulb device, and allowing the range to go beyond the range of individual bulbs in the group.  So using a specific config for this bulb that customises the range to 2200-5000 would actually make the problem worse.

**Reply by make-all** [reply]
it seems from the code, that light groups choose the minimum min_color_temp and maximum max_color_temp for the range of the group.  This suggests that maybe light integrations are expected to fail silently on out of range values.

**Reply by mirisbowring** [reply]
It worked before, but i replaced the entities in the light group instead of creating new ones - so probably this issue comes from this right?

**Reply by mirisbowring** [reply]
Nope created a new group and same bahviour.
I am just curious on why this behavious did not happend before? (with the loca-tuya integration)

**Reply by make-all** [reply]
Prior to a couple of versions ago, all tuya-local lightbulbs had the default HA color temp range of 2000-6500K.  Now the range can be set in the config file, and the generic configs are set to the more common 2700-6500 range.  This was to support a couple of requests for warm white bulbs with 1800-2700 ranges.

But to be getting a value of 6535 sent to the bulbs, as in the log posted, you must have a bulb in the group that goes higher than 6500K, and I don't think that is coming from any of the tuya-local configs.

**Reply by mirisbowring** [reply]
Mhm. Somehow it is only related to mushroom light cards... I used another light card and do not have the problem anymore.
o i would say "problem fixed" or "not related to tuya_local".

How can i change the color range in tuya_local? The UI don't seem to offer such thing :)

**Reply by make-all** [reply]
The color range is hard coded in the device config file (example: [gold_cct.lightbulb.yaml](https://github.com/make-all/tuya-local/blob/main/custom_components/tuya_local/devices/gold_cct_lightbulb.yaml#L24)), but it does not seem to affect the default UI visually, only in the range of values that are sent to the light.  And it seems that mushroom cards may be ignoring the range altogether.

**Reply by mirisbowring** [reply]
So should i add a 2200-5000K CCT Ledvance PR or should i stick to the current "default cct" since the actual values (10-1000) are the same and the changes are just cosmetic? :)
